napari/utils/_indexing.py

```diff
@@ -0,0 +1,67 @@
+from typing import Dict, Tuple
+
+import numpy as np
+import numpy.typing as npt
+
+
+def visible_items_in_slice(
+    index: Tuple[npt.NDArray[np.int_], ...], position_in_axes: Dict[int, int]
+) -> npt.NDArray[np.bool_]:
+    """
+    Return a boolean array indicating which items are visible in the current
+    view based on its indices and the current slice.
```

Comment:
```suggestion
def elements_in_slice(
    index: Tuple[npt.NDArray[np.int_], ...], position_in_axes: Dict[int, int]
) -> npt.NDArray[np.bool_]:
    """Mask elements from a multi-dimensional index not in a given slice.

    Some n-D operations may edit data that is not visible in the current slice.
    Given slice position information (as a dictionary mapping axis to index on that
    axis), this function returns a boolean mask for the possibly higher-dimensional
    multi-index so that elements not currently visible are masked out. The
    resulting multi-index can then be subset and used to index into a texture or
    other lower-dimensional view.
```

This took me a while to grok but it's a useful function, so I've expanded in the docstring, and also altered the name slightly. Let me know if you don't like me.

In the future, I think a better function would take in a valid indexing operation on the right (a tuple including `slice(None)` for visible axes) and intersect the indices. But I don't think we should worry about that now, I just wanted to get the idea out of my head. 😊